### PR TITLE
OSOE-796: Fix setup-sql-server GitHub Windows Larger Runner incompatibility in Lombiq.GitHub.Actions

### LINKED ISSUE #320: Fix setup-sql-server GitHub Windows Larger Runner incompatibility (OSOE-796)
When using [`setup-sql-server`](https://github.com/Lombiq/GitHub-Actions/tree/dev/.github/actions/setup-sql-server) under Linux or under Windows but on either a GitHub-hosted Standard Runner or a GitRunners one, it runs to completion without issues. However, executing it under a GitHub-hosted Windows Larger Runner produces this output, failing the workflow:

```
Chocolatey v2.2.2
Installing the following packages:
sql-server-2022
By installing, you accept licenses for the packages.

chocolatey-dotnetfx.extension v1.0.1 [Approved]
chocolatey-dotnetfx.extension package files install completed. Performing other installation steps.
 Installed/updated chocolatey-dotnetfx extensions.
 The install of chocolatey-dotnetfx.extension was successful.
  Software installed to 'C:\ProgramData\chocolatey\extensions\chocolatey-dotnetfx'

netfx-4.7.2 v4.7.2 [Approved]
netfx-4.7.2 package files install completed. Performing other installation steps.
Microsoft .NET Framework 4.7.2 or later is already installed.
 The install of netfx-4.7.2 was successful.
  Software install location not explicitly set, it could be in package or
  default install location of installer.

sql-server-2022 v16.0.1000.6 [Approved]
sql-server-2022 package files install completed. Performing other installation steps.
ERROR: The running command stopped because the preference variable "ErrorActionPreference" or common parameter is set to Stop: A system reboot is pending. You must restart Windows first before installing SQL Server
The install of sql-server-2022 was NOT successful.
Error while running 'C:\ProgramData\chocolatey\lib\sql-server-2022\tools\chocolateyinstall.ps1'.
 See log for details.

Chocolatey installed 2/3 packages. 1 packages failed.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Failures
 - sql-server-2022 (exited -1) - Error while running 'C:\ProgramData\chocolatey\lib\sql-server-2022\tools\chocolateyinstall.ps1'.
 See log for details.
```

See e.g. [this run](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/actions/runs/7755813572/job/21152505961#step:8:49).

This might be an issue with Chocolatey or something that we can fix from our script.

Opened https://github.com/actions/runner-images/issues/9277 about this but the maintainers closed it.

Once this is fixed, use a Larger Runner in [the `build-and-test-larger-runners` job's config in OSOCE](https://github.com/Lombiq/Open-Source-Orchard-Core-Extensions/blob/42f4cac0f65a58319737bb030fcf3718c43990e4/.github/workflows/build-and-test-windows.yml#L41).

[Jira issue](https://lombiq.atlassian.net/browse/OSOE-796)

### PR DESCRIPTION
**Pull request by Psichorex**
[OSOE-796](https://lombiq.atlassian.net/browse/OSOE-796)
Fixes #320

[OSOE-796]: https://lombiq.atlassian.net/browse/OSOE-796?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: .github/actions/setup-sql-server/Initialize-SqlServer.ps1 [hunk 1]
```diff
@@ -1,10 +1,11 @@
 param (
+    [string]$sqlServerExpressVersion = '2022.16.0.1000',
```
Comment (Piedone) [inline .github/actions/setup-sql-server/Initialize-SqlServer.ps1]: https://community.chocolatey.org/packages/sql-server-express says "Requires a reboot due to dependencies." How come it doesn't?

Comment (Piedone) [inline .github/actions/setup-sql-server/Initialize-SqlServer.ps1]:
```suggestion
    [string]$windowsSqlServerExpressVersion = '2022.16.0.1000',
```

File: .github/actions/setup-sql-server/Initialize-SqlServer.ps1 [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
     [string]$sqlServerVersion = '2022'
```
Comment (Piedone) [inline .github/actions/setup-sql-server/Initialize-SqlServer.ps1]:
```suggestion
    [string]$linuxSqlServerVersion = '2022'
```

**Review by Piedone — CHANGES_REQUESTED** [review]

File: .github/actions/setup-sql-server/Initialize-SqlServer.ps1 (hunk 1 above)
Comment (Psichorex) [reply .github/actions/setup-sql-server/Initialize-SqlServer.ps1]: I didnt look into that but we used sql express before osoe 481. So it is working for a while. I will try to check that.

Comment (Psichorex) [reply .github/actions/setup-sql-server/Initialize-SqlServer.ps1]: image
image
According to this there is no need to reboot.

Comment (Psichorex) [reply .github/actions/setup-sql-server/Initialize-SqlServer.ps1]: It more visible here:
image

Comment (Piedone) [reply .github/actions/setup-sql-server/Initialize-SqlServer.ps1]: This seems conflicting for me. Anyway, let's leave it like that.

File: .github/actions/setup-sql-server/Initialize-SqlServer.ps1 [hunk 3]
```diff
@@ -1,15 +1,16 @@
 param (
-    [string]$sqlServerVersion = '2022'
+    [string]$windowsSqlServerExpressVersion = '2022.16.0.1000',
+    [string]$linuxSqlServerVersion = '2022'
 )
 
 if ($Env:RUNNER_OS -eq 'Windows')
 {
-    choco install sql-server-$sqlServerVersion --no-progress
+    choco install sql-server-express --version=$windowsSqlServerExpressVersion--no-progress
```
Comment (BenedekFarkas) [inline .github/actions/setup-sql-server/Initialize-SqlServer.ps1]:
```suggestion
    choco install sql-server-express --version=$windowsSqlServerExpressVersion --no-progress
```

**Review by Piedone — APPROVED** [review]

**Pull request merged**